### PR TITLE
fix: correctly save block orientations in structure schematics

### LINKED ISSUE #3028: Vanilla Schematic doesn't save blockstates correctly
### Server Implementation

Paper

### Server Version

1.21.3

### Describe the bug

When saving a schematic as a vanilla .nbt file, the blockstates are not saved.

### To Reproduce

1. Select area with blocks that have blockstates (such as stairs)
2. Copy and create a vanilla schematic
3. Move schematic to a datapack
4. run /place template (your schematic)

### Expected behaviour

Blocks should be placed in with the correct blockstates

### Screenshots / Videos

image

### Error log (if applicable)

_No response_

### Fawe Debugpaste

https://athion.net/ISPaster/paste/view/3b8dc36420bc48049d7b0a3feea316ed

### Fawe Version

FastAsyncWorldEdit-955

### Checklist

- [X] I have included a Fawe debugpaste.
- [X] I am using the newest build from https://ci.athion.net/job/FastAsyncWorldEdit/ and the issue still persists.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by PierreSchwang**
## Overview
Fixes #3028

## Description
- Fixes the linked issue
- Migrate to lin-bus from jnbt
- use datafixer if applicable

```[tasklist]
### Submitter Checklist
- [x] Make sure you are opening from a topic branch (**/feature/fix/docs/ branch** (right side)) and not your main branch.
- [x] Ensure that the pull request title represents the desired changelog entry.
- [x] New public fields and methods are annotated with `@since TODO`.
- [x] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/.github/blob/main/CONTRIBUTING.md).
```

### CODE REVIEW

File: worldedit-core/src/main/java/com/fastasyncworldedit/core/extent/clipboard/io/schematic/MinecraftStructure.java [hunk 1]
```diff
@@ -1,273 +1,291 @@
...
+                continue;
+            }
+            clipboard.setBlock(
+                    pos.get(0).valueAsInt(), pos.get(1).valueAsInt(), pos.get(2).valueAsInt(),
+                    palette[state].toBaseBlock(nbt)
+            );
+        }
+
+        // Entities
+        LinListTag<@org.jetbrains.annotations.NotNull LinCompoundTag> entities = parent.findListTag("entities", LinTagType.compoundTag());
```
Comment (SirYwell) [inline worldedit-core/src/main/java/com/fastasyncworldedit/core/extent/clipboard/io/schematic/MinecraftStructure.java:155]: Can you switch to the javax Nonnull annotation here? That's already imported

Comment (PierreSchwang) [reply worldedit-core/src/main/java/com/fastasyncworldedit/core/extent/clipboard/io/schematic/MinecraftStructure.java:155]: Sadly not really possible, as for the javax annotations, generic types are not set as an allowed target. The only reason I've annotated them, is to get rid of the warning, as lin enforces nonnull there

**Review by SirYwell — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]

File: worldedit-core/src/main/java/com/fastasyncworldedit/core/extent/clipboard/io/schematic/MinecraftStructure.java [hunk 2]
```diff
@@ -1,273 +1,291 @@
...
         return clipboard;
     }
 
     @Override
     public void write(Clipboard clipboard) throws IOException {
         clipboard.flush();
-        write(clipboard, "FAWE");
-    }
-
-    public void write(Clipboard clipboard, String owner) throws IOException {
```
Comment (dordsor21) [inline worldedit-core/src/main/java/com/fastasyncworldedit/core/extent/clipboard/io/schematic/MinecraftStructure.java:159]: Is there a particular reason for removing owner? We should also have it as a deprecated method still to ensure API compatibility

Comment (PierreSchwang) [reply worldedit-core/src/main/java/com/fastasyncworldedit/core/extent/clipboard/io/schematic/MinecraftStructure.java:159]: > author: Name of the player who created this structure. Only exists for structures saved before 1.13.

https://minecraft.wiki/w/Structure_file


Will add method again and delegate to `write(Clipboard)`

**Review by dordsor21 — APPROVED** [review]

**Pull request merged**